### ISSUE TITLE
Issues trying to use BayesianCV on an xgb classifier estimator

### ISSUE THREAD

**Issue by PyDataBlog** [scikit-optimize/scikit-optimize#779]
I seem to have stumbled into some bug trying to do a Bayesian CV with an XGB Classifier estimator. My skopt version is `0.5.2` and my xgb version is `0.90`. Here's the code and the error output;

```py
from skopt import BayesSearchCV

# include below until https://github.com/scikit-optimize/scikit-optimize/issues/718 is resolved
class BayesSearchCV(BayesSearchCV):
    def _run_search(self, x): raise BaseException('Use newer skopt')
        
xgb_opt_all = XGBClassifier()

param_xgb = {
    'base_score': (0.01, 0.99, 'uniform'),
    'learning_rate': (0.01, 1, 'uniform'),
}

xgb_grid = BayesSearchCV(
    estimator=xgb_opt_all, search_spaces=param_xgb,
    scoring='accuracy', n_jobs=-1, cv=10)
```

```py
The error output is like this:
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-53-b1c76ec62c4e> in <module>
     14 xgb_grid = BayesSearchCV(
     15     estimator=xgb_opt_all, search_spaces=param_xgb,
---> 16     scoring='accuracy', n_jobs=-1, cv=10)

~\AppData\Local\Continuum\anaconda3\envs\classifier-v2\lib\site-packages\skopt\searchcv.py in __init__(self, estimator, search_spaces, optimizer_kwargs, n_iter, scoring, fit_params, n_jobs, n_points, iid, refit, cv, verbose, pre_dispatch, random_state, error_score, return_train_score)
    295              n_jobs=n_jobs, iid=iid, refit=refit, cv=cv, verbose=verbose,
    296              pre_dispatch=pre_dispatch, error_score=error_score,
--> 297              return_train_score=return_train_score)
    298 
    299     def _check_search_space(self, search_space):

TypeError: __init__() got an unexpected keyword argument 'fit_params'

```

Am I doing something wrong or is this an upstream bug?

**Comment by cl445** [comment]
got the same issue independent from the used classification algorithm. 
Maybe it is related to the new scikit-learn version 0.21?

**Comment by cl445** [comment]
Okay, BaseSearch, the super class from BayesSearchCV has not the parameter 'fit_params', but it is called in the constructor of BayesSearchCV. So this is a serious bug!

**Comment by PyDataBlog** [comment]
@cl445 unfortunately, it seems to look like a serious bug that I have stumbled upon.

**Comment by rkingery** [comment]
Really wish somebody would fix this issue with the sklearn 0.21 version. The main purpose of this whole library to me and many others--hyperparameter optimization--is basically unusable without that support. Such a shame development effort on this library has decreased recently, because hyperopt is so much more annoying to use with sklearn IMO. Moving onto bigger and better things I guess...

**Comment by PyDataBlog** [comment]
@rkingery any other project out there?

**Comment by gwerbin** [comment]
I believe this is a one-line fix https://github.com/scikit-optimize/scikit-optimize/blob/80d8a7b08c870bb032f06d5e4c34794c10376fd7/skopt/searchcv.py#L293-L297:

Just remove `fit_params` from the kwargs in the `super()` call.

I'm busy with work at the moment, but hopefully someone is able to quickly make the change and test. Otherwise I will try to convince my team that this is worth fixing on the clock (since we use BayesSearchCV in our work).

**Comment by wud666** [comment]
> I believe this is a one-line fix:
> 
> https://github.com/scikit-optimize/scikit-optimize/blob/80d8a7b08c870bb032f06d5e4c34794c10376fd7/skopt/searchcv.py#L293-L297
> 
> Just remove `fit_params` from the kwargs in the `super()` call whenever the sklearn version is >= 0.21 (which can be determined at module-load time).
> 
> I'm busy with work at the moment, but hopefully someone is able to quickly make the change and test. Otherwise I will try to convince my team that this is worth fixing on the clock (since we use BayesSearchCV in our work).
> 
> Edit: someone apparently has a fork with this change in place [#762 (comment)](https://github.com/scikit-optimize/scikit-optimize/issues/762#issuecomment-493689266)

remove fit_params would not solve this problem since codes below also use this parameter

**Comment by glouppe** [comment]
Fixed in #777
